docs/DeveloperGuide.md

```diff
@@ -411,18 +452,47 @@ The tradeoffs for this approach is examined below:
       * This may lead to unexpected behaviours, such as whether properties linked to an owner should be deleted when
       the owner is deleted.
 
-### Filtering properties by price range
+### Filtering buyers and properties: Overall structure
 
-The `Properties` list is filtered using a predicate, `filterPropsByPricePredicate`. This predicate checks if the
-property's price falls within a specified price range.
+In order to filter `Buyers` and `Properties`, a `Predicate` needs to be passed into the `ObservableArrayList` that stores 
+references to these objects and displays them on the user's screen. These predicates can differ in the conditions that are
+being tested, consequently, they might give different outputs when applied to a given list.
 
-The structure for executing a `filterprops` command follows the flow as mentioned in the “Logic component” section of
-this guide.
+#### Design Considerations
+In order to allow for multiple-condition filtering, that is, the composition of multiple filter predicates, an abstract 
+`AbstractFilterXYZPredicate` class was created to employ polymorphic behaviour, where XYZ represents the entry type that
+we are working with, for example `AbstractFilterBuyerPredicate` or `AbstractFilterPropsPredicate`. 
 
-#### Design considerations:
-As `Property` has a single specific `Price`, it is much less useful to filter the list using one price value as it is
-unlikely to match any property. Instead, we decided to filter by a price range instead, where any property whose price
-falls within this range would be displayed.
+Additionally, since users
+are only allowed to filter using certain conditions as defined in the behaviour of the `filter` commands, concrete classes
+extending this abstract predicate class were implemented for each condition. For example:
+
+Users can filter `Properties` by their `PropertyName`, `Price`, `Characteristics` or `OwnerName`. As a result, the following 
+concrete predicate classes were implemented:
+1. `PropertyNameContainsSubstringPredicate`
+2. `FilterPropsByPricePredicate`
+3. `FilterPropsContainingAllCharacteristicsPredicate`
+4. `FilterPropsContainingAnyCharacteristicsPredicate`
+5. `FilterPropsByOwnerNamePredicate`
+
+Based on command parameters passed in by the user, these predicates are constructed and composed together to form a single
+`Predicate`, which is then used to filter the `ObservableArrayList` directly.
```

Comment:
Would be clearer to mention here that there are really only 2 ways:
1. We compose using OR if the user pass in -fuzzy
2. We compose using AND as default